app/src/main/java/com/example/randomusergenerator/user/viewmodel/UserViewModel.kt

```diff
@@ -2,33 +2,35 @@ package com.example.randomusergenerator.user.viewmodel
 
 import androidx.lifecycle.ViewModel
 import androidx.lifecycle.viewModelScope
-import com.example.randomusergenerator.data.local.UserData
+import com.example.randomusergenerator.data.local.UserViewState
 import com.example.randomusergenerator.domain.repository.UserRepository
 import com.example.randomusergenerator.utils.Resource
 import dagger.hilt.android.lifecycle.HiltViewModel
 import kotlinx.coroutines.flow.MutableStateFlow
 import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.update
 import kotlinx.coroutines.launch
 import javax.inject.Inject
 
 @HiltViewModel
 class UserViewModel @Inject constructor(private val userRepository: UserRepository) : ViewModel() {
-    private var _userResponse = MutableStateFlow<Resource<List<UserData>>?>(null)
-    val userResponse: StateFlow<Resource<List<UserData>>?> get() = _userResponse
+    private var _uiState = MutableStateFlow(UserViewState())
+    val uiState: StateFlow<UserViewState> get() = _uiState
 
-    init {
-        getAllUsers(10)
```

Comment:
Why did you remove this function and the `init{}` block?

Reply:
I wanted to call the functions in the test class. It makes it readable for me

Reply:
Oh! So, you will need to call the function in the view as well, right?

Reply:
yes. I will do that